src/modules/rlm_smtp/rlm_smtp.c

```diff
@@ -0,0 +1,1107 @@
+	tmpl_t 			**attachments;		//!< The attachmen ts to be set
+	tmpl_t 			**recipient_addrs;	//!< Comma separated list of emails. Overrides elements in to, cc, bcc
+	tmpl_t 			**to_addrs;		//!< Comma separated list of emails to be listed in TO:
+	tmpl_t 			**cc_addrs;		//!< Comma separated list of emails to be listed in CC:
+	tmpl_t 			**bcc_addrs;		//!< Comma separated list of emails not to be listed
+	fr_time_delta_t 	timeout;		//!< Timeout for connection and server response
+	fr_curl_tls_t		tls;			//!< Used for handled all tls specific curl components
+	char const		*name;			//!< Auth-Type value for this module instance.
+	fr_dict_enum_t		*auth_type;
+	vp_map_t		*header_maps;			//!< Attribute map used to process header elements
+	CONF_SECTION		*cs;
+	bool 			set_date;
+} rlm_smtp_t;
+
+typedef struct {
+	rlm_smtp_t const    	*inst;		//!< Instance of rlm_smtp.
+	fr_curl_handle_t    	*mhandle;	//!< Thread specific multi handle.  Serves as the dispatch and coralling structure for smtp requests
+} rlm_smtp_thread_t;
+
+/*
+ *	Holds the context for parsing the email elements
+ */
+typedef struct {
+	REQUEST			*request;
+	fr_curl_io_request_t	*randle;
+	fr_cursor_t		cursor;
+	fr_cursor_t		body_cursor;
+	fr_dbuff_t		vp_in;
+	struct curl_slist	*recipients;
+	struct curl_slist	*header;
+	struct curl_slist 	*body_header;
+	fr_time_t 		time;
+	char 			time_str[60];
+	curl_mime		*mime;
+} fr_mail_ctx;
+
+/*
+ *	A mapping of configuration file names to internal variables.
+ */
+static const CONF_PARSER module_config[] = {
+	{ FR_CONF_OFFSET("uri", FR_TYPE_STRING, rlm_smtp_t, uri) },
+	{ FR_CONF_OFFSET("template_directory", FR_TYPE_STRING, rlm_smtp_t, template_dir) },
+	{ FR_CONF_OFFSET("attachments", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, attachments), .dflt = "&SMTP-Attachments", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("sender_address", FR_TYPE_STRING, rlm_smtp_t, sender_address) },
+	{ FR_CONF_OFFSET("envelope_address", FR_TYPE_STRING, rlm_smtp_t, envelope_address) },
+	{ FR_CONF_OFFSET("recipients", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, recipient_addrs), .dflt = "&SMTP-Recipients", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("TO", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, to_addrs), .dflt = "&SMTP-TO", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("CC", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, cc_addrs), .dflt = "&SMTP-CC", .quote = T_BARE_WORD},
+	{ FR_CONF_OFFSET("BCC", FR_TYPE_TMPL | FR_TYPE_MULTI, rlm_smtp_t, bcc_addrs), .dflt = "&SMTP-BCC", .quote = T_BARE_WORD },
+	{ FR_CONF_OFFSET("timeout", FR_TYPE_TIME_DELTA, rlm_smtp_t, timeout) },
+	{ FR_CONF_OFFSET("set_date", FR_TYPE_BOOL, rlm_smtp_t, set_date), .dflt = "yes" },
+	{ FR_CONF_OFFSET("tls", FR_TYPE_SUBSECTION, rlm_smtp_t, tls), .subcs = (void const *) fr_curl_tls_config },//!<loading the tls values
+	CONF_PARSER_TERMINATOR
+};
+
+/** Iterate over a talloced array of elements
+ *
+@verbatim
+talloc_foreach(vpt_m, tmpl_t *, vpt) {
+	tmpl_debug(vpt);
+}
```

Comment:
That's defined centrally.  Just include the `<freeradius-devel/talloc.h>` header